docs(redis) update redis chapter (#735)
nestjs/docs.nestjs.com: content/microservices/redis.md

### Redis

> [Redis](https://redis.io/) transporter implements the publish/subscribe messaging paradigm and leverages [Pub/Sub](https://redis.io/topics/pubsub) feature of Redis. Published messages are categorised in channels, without knowing what subscribers will eventually receive the message (and if any). Each microservice can subscribe to any number of channels. In addition, more than one channel can be subscribed to at a time. Messages exchanged through channels are **fire-and-forget**, which means that if a message is published and there are no subscribers interested in it, the message removed and cannot be recovered. Thus, you don't have a guarantee that either messages or events will be handled by at least one service. Also, single message can be received many subscribers.

Reviewer: The [Redis](https://redis.io/) transporter implements the publish/subscribe messaging paradigm and leverages the [Pub/Sub](https://redis.io/topics/pubsub) feature of Redis. Published messages are categorized in channels, without knowing what subscribers (if any) will eventually receive the message. Each microservice can subscribe to any number of channels. In addition, more than one channel can be subscribed to at a time. Messages exchanged through channels are **fire-and-forget**, which means that if a message is published and there are no subscribers interested in it, the message is removed and cannot be recovered. Thus, you don't have a guarantee that either messages or events will be handled by at least one service. A single message can be subscribed to (and received) by multiple subscribers.

> [Redis](https://redis.io/) transporter implements the publish/subscribe messaging paradigm and leverages [Pub/Sub](https://redis.io/topics/pubsub) feature of Redis. Published messages are categorised in channels, without knowing what subscribers will eventually receive the message (and if any). Each microservice can subscribe to any number of channels. In addition, more than one channel can be subscribed to at a time. Messages exchanged through channels are **fire-and-forget**, which means that if a message is published and there are no subscribers interested in it, the message removed and cannot be recovered. Thus, you don't have a guarantee that either messages or events will be handled by at least one service. Also, single message can be received many subscribers.

Author: ```suggestion
[Redis](https://redis.io/) transporter implements the publish/subscribe messaging paradigm and leverages [Pub/Sub](https://redis.io/topics/pubsub) feature of Redis. Published messages are categorized in channels, without knowing what subscribers (if any) will eventually receive the message. Each microservice can subscribe to any number of channels. In addition, more than one channel can be subscribed to at a time. Messages exchanged through channels are **fire-and-forget**, which means that if a message is published and there are no subscribers interested in it, the message is removed and cannot be recovered. Thus, you don't have a guarantee that either messages or events will be handled by at least one service. A single message can be subscribed to (and received) by multiple subscribers.
```

<figure><img src="/assets/Redis_1.png" /></figure>

#### Installation

To start building Redis-based microservices, first install the required package:

```bash
$ npm i --save redis
```

#### Overview

> In order to use Redis transporter, a different options object has to be passed to the `createMicroservice()` method.

Reviewer: To use the Redis transporter, pass the following options object to the `createMicroservice()` method:

> In order to use Redis transporter, a different options object has to be passed to the `createMicroservice()` method.

Author: ```suggestion
To use the Redis transporter, pass the following options object to the `createMicroservice()` method:
```

});
```

> > info **Hint** `Transport` enum is imported from the `@nestjs/microservices` package.

Reviewer: The `Transport` enum... <- add *The*

> > info **Hint** `Transport` enum is imported from the `@nestjs/microservices` package.

Author: ```suggestion
> info **Hint** The `Transport` enum is imported from the `@nestjs/microservices` package.
```

> Likewise, to create a client instance, we need to pass the options object which has the same properties we saw above in the `createMicroservice()` method.

Reviewer: Likewise, to create a client instance, we need pass an options object with the same properties we saw above in the `createMicroservice()` method.

> Likewise, to create a client instance, we need to pass the options object which has the same properties we saw above in the `createMicroservice()` method.

Author: ```suggestion
Likewise, to create a client instance, we need to pass an options object with the same properties we saw above in the `createMicroservice()` method.
```

```typescript
ClientsModule.register([
  {
    name: 'MATH_SERVICE',
    transport: Transport.REDIS,
    options: {
      url: 'redis://localhost:6379',
    }
  },
]),
```

Other options to create a client (either `ClientProxyFactory` or `@Client()`) can be used as well. You can read about them [here](/microservices/basics#client).

#### Context

> In more sophisticated scenarios, you may want to access more information about the incoming request. In Redis, you can access `RedisContext` object.

Author: ```suggestion
In more sophisticated scenarios, you may want to access more information about the incoming request. In Redis, you can access the `RedisContext` object.
```

> In more sophisticated scenarios, you may want to access more information about the incoming request. In Redis, you can access `RedisContext` object.

Reviewer: ...you can access the `ReisContext`... <- add *the*

  </tr>
  <tr>
    <td><code>retryAttempts</code></td>

>     <td>Number of times to retry messag</td>

Reviewer: message <- fix typo

>     <td>Number of times to retry messag</td>

Author: ```suggestion
    <td>Number of times to retry message</td>
```
